astropy/stats/spatial.py

```diff
@@ -0,0 +1,307 @@
+
+            for r in range(len(radii)):
+                dist_indicator = distances < radii[r]
+                ripley[r] = ((1 / intersec_area) * dist_indicator).sum()
+
+            ripley = (self.area**2 / (npts * (npts - 1))) * 2 * ripley
+        # Stoyan book page 123 and eq 15.13
+        elif mode == 'ohser':
+            diff = self._pairwise_diffs(data)
+            distances = np.hypot(diff[:, 0], diff[:, 1])
+            a = self.area
+            b = max((self.y_max - self.y_min) / (self.x_max - self.x_min),
+                    (self.x_max - self.x_min) / (self.y_max - self.y_min))
+            x = distances / math.sqrt(a / b)
+            u = np.sqrt((x * x - 1) * (x > 1))
+            v = np.sqrt((x * x - b ** 2) * (x < math.sqrt(b ** 2 + 1)) * (x > b))
+            c1 = np.pi - 2 * x * (1 + 1 / b) + x * x / b
+            c2 = 2 * np.arcsin((1 / x) * (x > 1)) - 1 / b - 2 * (x - u)
+            c3 = (2 * np.arcsin(((b - u * v) / (x * x))
+                                * (x > b) * (x < math.sqrt(b ** 2 + 1)))
+                  + 2 * u + 2 * v / b - b - (1 + x * x) / b)
+
+            cov_func = ((a / np.pi) * (c1 * (x >= 0) * (x <= 1)
+                        + c2 * (x > 1) * (x <= b)
+                        + c3 * (b < x) * (x < math.sqrt(b ** 2 + 1))))
+
+            for r in range(len(radii)):
+                dist_indicator = distances < radii[r]
+                ripley[r] = ((1 / cov_func) * dist_indicator).sum()
+
+            ripley = (self.area**2 / (npts * (npts - 1))) * 2 * ripley
+        # Cressie book eq 8.2.20 page 616
+        elif mode == 'var-width':
+            lt_dist = np.zeros(npts, dtype=np.double)
+            for k in range(npts):
+                lt_dist[k] = min(self.x_max - data[k][0],
+                                 self.y_max - data[k][1],
+                                 data[k][0] - self.x_min,
+                                 data[k][1] - self.y_min)
+
+            for r in range(len(radii)):
+                for i in range(npts):
+                    for j in range(npts):
+                        if i != j:
+                            diff = abs(data[i] - data[j])
+                            dist = math.sqrt((diff * diff).sum())
+                            if dist < radii[r] and lt_dist[i] > radii[r]:
+                                ripley[r] = ripley[r] + 1
+                if not (lt_dist > radii[r]).sum() == 0:
+                    ripley[r] = ripley[r] / (lt_dist > radii[r]).sum()
+
+            ripley = self.area * ripley / npts
+        # Cressie book eq 8.4.22 page 640
+        elif mode == 'ripley':
+            hor_dist = np.array([])
+            ver_dist = np.array([])
+
+            for k in range(npts - 1):
+                min_hor_dist = min(self.x_max - data[k][0], data[k][0] - self.x_min)
+                min_ver_dist = min(self.y_max - data[k][1], data[k][1] - self.y_min)
+                hor_dist = np.append(hor_dist, min_hor_dist * np.ones(npts - 1 - k))
```

Comment:
Instead of `append` you could also initialize the final-sized array before and just set the slices. The values should be known, right?